pandas/core/internals/blocks.py

```diff
@@ -839,6 +839,9 @@ def setitem(self, indexer, value):
 
         # coerce if block dtype can store value
         values = self.values
```

Comment:
If we do end up needing this special case, it might make sense to put this in ObjectBlock.setitem; not a strong opinion